### PR TITLE
Transmute refactor

### PR DESCRIPTION
**Pull request by mxcl**
Preparing for `--edit` implementation

### CODE REVIEW

**Comment by mxcl** [comment]
@swift-ci Please test and merge

**Comment by mxcl** [comment]
@swift-ci Please test and merge

**Comment by mxcl** [comment]
@swift-ci Please test and merge

File: Sources/Multitool/Options.swift [hunk 1]
```diff
@@ -26,7 +26,8 @@ public class Options {
             set { _build = newValue }
         }
 
-        private var _build = getenv("SWIFT_BUILD_PATH")
+        private var _Packages: String?
+            private var _build: String?
```
Comment (kostiakoval) [inline Sources/Multitool/Options.swift]: wrong indent

File: Sources/Multitool/Options.swift [hunk 2]
```diff
@@ -26,7 +26,8 @@ public class Options {
             set { _build = newValue }
         }
 
-        private var _build = getenv("SWIFT_BUILD_PATH")
+        private var _Packages: String?
```
Comment (kostiakoval) [inline Sources/Multitool/Options.swift]: lowercase name for variables. `packages`

Comment (mxcl) [inline Sources/Multitool/Options.swift]: It represents the `Packages` directory.

File: Sources/Multitool/Options.swift (hunk 1 above)
Comment (mxcl) [inline Sources/Multitool/Options.swift]: Fixed.

**Comment by mxcl** [comment]
@swift-ci Please test and merge

File: Sources/Multitool/Options.swift (hunk 2 above)
Comment (kostiakoval) [inline Sources/Multitool/Options.swift]: Aha, ok then :) 
I also noticed that `_Packages` is never used. Is is suppose to be so or should it be use in 
`public var Packages: String` ?

**Comment by mxcl** [comment]
@swift-ci Please test

**Comment by mxcl** [comment]
@ddunbar you ok with this?

**Comment by mxcl** [comment]
@swift-ci Please test

File: Sources/Transmute/Apparatus.swift [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import PackageType
+import Utility
+
+
+class Apparatus {
```
Comment (ddunbar) [inline Sources/Transmute/Apparatus.swift]: What is the purpose of this class? Can we get a doc-comment here?

File: Tests/Functional/Utilities.swift [hunk 4]
```diff
@@ -118,21 +118,7 @@ func executeSwiftBuild(_ chdir: String, configuration: Configuration = .Debug, p
...
-            swiftc = Path.join(base, "usr/bin/swiftc")
-        } else {
-            swiftc = try popen(["xcrun", "--find", "swiftc"]).chuzzle() ?? "BADPATH"
-        }
-        precondition(swiftc != "/usr/bin/swiftc")
-        env["SWIFT_EXEC"] = swiftc
-    default:
-        fatalError("HURRAY! This is fixed")
-    }
+    env["SWIFT_EXEC"] = figure_out_SWIFT_EXEC()
```
Comment (ddunbar) [inline Tests/Functional/Utilities.swift]: I pushed a different fix for this last night, we will now respect TOOLCHAINS in the environment... if you set TOOLCHAINS for the tests (either by launching Xcode with it, or by changing the environment of the scheme) then that works.

Let's chat today to figure out if there is more we need to do here...

**Comment by ddunbar** [comment]
Sorry, I missed the question here... the changes seem ok to me but I'm not clear what conceptual role `Apparatus` fills yet. Also, I pushed a different fix for the `TOOLCHAINS` issue (although it might not be as complete as this one).

**Comment by ddunbar** [comment]
Closing this out while we figure out the right answer here...

**Pull request closed without merging**